app/javascript/components/character-count.js

```diff
@@ -0,0 +1,187 @@
+  var attributes = element.attributes
+  if (attributes) {
+    for (var i = 0; i < attributes.length; i++) {
+      var attribute = attributes[i]
+      var match = attribute.name.match(/^data-(.+)/)
+      if (match) {
+        dataset[match[1]] = attribute.value
+      }
+    }
+  }
+  return dataset
+}
+
+// Counts characters or words in text
+CharacterCount.prototype.count = function (text, options) {
+  var length
+  if (options.maxwords) {
+    var tokens = text.match(/\S+/g) || [] // Matches consecutive non-whitespace chars
+    length = tokens.length
+  } else {
+    length = text.length
+  }
+  return length
+}
+
+// Generate count message and bind it to the input
+// returns reference to the generated element
+CharacterCount.prototype.createCountMessage = function () {
+  var countElement = this.$module
+  var elementId = countElement.id
+  // Check for existing info count message
+  var countMessage = document.getElementById(elementId + '-info')
+  // If there is no existing info count message we add one right after the field
+  if (elementId && !countMessage) {
+    countElement.insertAdjacentHTML('afterend', '<span id="' + elementId + '-info" class="govuk-hint govuk-character-count__message" aria-live="polite"></span>')
+    this.describedBy = countElement.getAttribute('aria-describedby')
+    this.describedByInfo = this.describedBy + ' ' + elementId + '-info'
+    countElement.setAttribute('aria-describedby', this.describedByInfo)
+    countMessage = document.getElementById(elementId + '-info')
+  }
+  return countMessage
+}
+
+// Bind input propertychange to the elements and update based on the change
+CharacterCount.prototype.bindChangeEvents = function () {
+  var $module = this.$module
+  $module.addEventListener('keyup', this.updateCountMessage.bind(this))
+
+  // Bind focus/blur events to start/stop polling
+  $module.addEventListener('focus', this.handleFocus.bind(this))
+  $module.addEventListener('blur', this.handleBlur.bind(this))
+}
+
+// Speech recognition software such as Dragon NaturallySpeaking will modify the
+// fields by directly changing its `value`. These changes don't trigger events
+// in JavaScript, so we need to poll to handle when and if they occur.
+CharacterCount.prototype.checkIfValueChanged = function () {
+  if (!this.oldValue) this.oldValue = ''
+  if (this.$module.value !== this.oldValue) {
+    this.oldValue = this.$module.value
+    var boundUpdateCountMessage = this.updateCountMessage.bind(this)
+    boundUpdateCountMessage()
+  }
+}
+
+// Update message box
+CharacterCount.prototype.updateCountMessage = function () {
+  var countElement = this.$module
+  var options = this.options
+  var countMessage = this.countMessage
+
+  // Determine the remaining number of characters/words
+  var currentLength = this.count(countElement.value, options)
+  var maxLength = this.maxLength
+  var remainingNumber = maxLength - currentLength
+
+  // Set threshold if presented in options
+  var threshold = 0
```

Comment:
You could make this a ternary - it seems to be the pattern used in the code above.